src/v/cluster/partition.cc

```diff
@@ -188,12 +188,12 @@ model::offset partition::start_cloud_offset() const {
       _cloud_storage_partition->first_uploaded_offset());
 }
 
-model::offset partition::last_cloud_offset() const {
+model::offset partition::next_cloud_offset() const {
```

Comment:
should it return`kafka::offset`?

Reply:
It should, but for the sake of keeping this PR small, I'll punt and do this in a follow-up.